lib/core_extensions.rb

```diff
@@ -39,8 +39,14 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
+        record.association(klass.to_sym).association_scope.each do |what|
+          unless what.is_a? String
+            if what.class.find_by_id(what.id)
```

Comment:
What's the intention of this?  To see if the object is in the default scope or not, and to deliberately give only the class name if it's out of scope?